net/connUDP.go

```diff
@@ -24,10 +24,36 @@ type UDPConn struct {
 }
 
 type ControlMessage struct {
+	Dst     net.IP // destination address, receiving only
 	Src     net.IP // source address, specifying only
```

Comment:
Source-Address is also valid when receiving packet, same address as the "remote" parameter you get from the read

Reply:
According to doc the https://pkg.go.dev/golang.org/x/net@v0.18.0/ipv4#ControlMessage it seems that `Src` cannot be set for write.

Reply:
doc says "specifying only" for source (as in "write only).

but I am quite sure (just tested it on Debian) that you ALSO get the source set when receiving.